### ISSUE TITLE
[WSL 1] (2.2.3.0 (43965) edge) Host mount paths incorrectly displayed, I/O issues

### ISSUE THREAD

**Issue by strarsis** [docker/for-win#6209]
- [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: `DB6939B5-5158-41B4-8EA0-9DC8A82F5590/20200402200807`

### Expected behavior
As with the previous Docker edge (and stable) releases, the path in WSL where the `docker-compose.yml` is located and from where `docker-compose` is invoked, should be correctly translated to the Windows Host path and I/O should still work as before so that containers like mysql/mariadb can start/initialize correctly.

### Actual behavior
WSL cwd is on a directory located on drive D: on Windows Host system.
Although drive D: has been added as shared drive, on `docker-compose up -d` Docker prompts for allowing a share. The path on the host system displayed in that prompt is not the correct Windows path, instead it is the untranslated, original WSL path with `C:\` prepended. Allowing the share seems to still mount the directory, but some containers fail to start up/initialize now, like mariadb/mysql.
````
[ERROR] InnoDB: preallocating 12582912 bytes for file ./ibdata1 failed with error 38
[ERROR] InnoDB: Could not set the file size of './ibdata1'. Probably out of disk space
````
This hadn't been an issue with previous Docker stable/edge versions.
Start up command: `mysqld --innodb-flush-method=littlesync --innodb-use-native-aio=OFF --log_bin=ON   --skip-innodb-file-per-table`

### Information
- Is it reproducible?: Yes
- Is the problem new?: Yes
- Did the problem appear with an update?: Yes, the update from `2.2.2.0 (43066) edge` to `2.2.3.0 (43965) edge` as it was offered today.
  - Windows Version: `Version	10.0.18363 Build 18363`
  - Docker Desktop Version: `2.2.3.0 (43965) edge`
  - Are you running inside a virtualized Windows e.g. on a cloud server or on a mac VM: No, Hyper-V on native Windows 10 x64 Pro.

Downgrading back to `2.2.2.0 (43066) edge` resolved this issue.

**Comment by stephen-turner** [comment]
Thanks for the report, @strarsis, looking at it.

**Comment by stephen-turner** [comment]
Assigning to @ebriney.

**Comment by ebriney** [comment]
Hi @strarsis, there is probably an unhandled case in the directory authorizer.

Which path is displayed in the notification windows?

What did you put in compose file `.`?

**Comment by strarsis** [comment]
@ebriney: Thanks for looking into this! This new release fixes an [I/O issue with some golang programs](https://github.com/jwilder/docker-gen/issues/317) but introduces this one. 😸 

I will upgrade again to reproduce the issue and report back.

**Comment by strarsis** [comment]
@ebriney: OK, so I upgraded again from Docker Desktop `2.2.2.0 (43066)` to Docker Desktop `2.2.3.0 (43965)`.
It is possible to use the upgraded Docker Desktop version without rebooting Windows (and no reboot is prompted by the installer). Everything starts up again and now Docker Desktop `2.2.3.0 (43965)` is running. I emptied all the preset shares.

In WSL 1 (Ubuntu) I open a simple WordPress app `docker-compose` project and do a `docker-compose up -d`.
The share prompt opens and to my surprise, the path is correct compared to what I see the last time.
The directory actually to be mounted into the container got the path `/d/DATA/[...]`, 
the prompt shows the path `D:\d\DATA\[...]`, so the `D:\` is redundant but somehow correct as it is indeed disk with drive letter `D:` on the host system.
I `down`ed the container setup and rebooted the system because I suspect the incorrect path as described for this issue indeed only shows up after a reboot.

After Windows reboot, the share prompt now indeed shows the (more) incorrect path:
`C:\d\DATA\[...]` instead of the nearly correct `D:\d\DATA\[...]` before the reboot for the actual path `/d/DATA/[...]` in WSL.
The reboot seems to reload/change something that results in a more incorrect path to be shown by the prompt.

But not only the path in share prompt is incorrect, I/O doesn't work correctly either after upgrading Docker Desktop: The containers can't read/write from/to the mounted host directories anymore.
Trying to list the mounted volume directory from within such a container results in an error:
`dir: reading directory '.': Operation not permitted`. The containers can't initialize correctly, no config or data files can be read anymore.

It seems that the incorrect path shown by share prompt and the permission issues are somehow related.

**Comment by ebriney** [comment]
@strarsis Thanks for the details. Yes, permission and authorized directories are linked. We have a ticket to address the issue and will work for a fix ASAP.

**Comment by strarsis** [comment]
@ebriney: This is great! Is a release planned in the next days or next week?

**Comment by stephen-turner** [comment]
This Edge is RC1 for the next Stable (2.3), but we don't have a firm release date for Stable yet. We need to work through the bugs that have been reported on this Edge release.

**Comment by ebriney** [comment]
Hi @strarsis , can you test that the fix in this build is working for you:  https://download-stage.docker.com/win/edge/44301/Docker%20Desktop%20Installer.exe

**Comment by strarsis** [comment]
@ebriney: This would be the previous version (`2.2.4.0`) I already downgraded to on which the I/O error doesn't yet occur.

**Comment by ebriney** [comment]
@strarsis ??? you have downgraded to `2.2.2.0`
This one is not out at all, it's a PR build.
Anyway, new edge is nearly ready and stable should follow next week including this patch.

**Comment by strarsis** [comment]
@ebriney: Ah, I see. The build you linked to (from PR) got version `2.2.4.0` in its metadata, 
the build process should add the PR name / short commit SHA to the version in metadata.

**Comment by strarsis** [comment]
@ebriney: So I installed the latest Docker Edge 2.3.00 (44472).
The mariadb database container doesn't start after the update.
The mariadb server fails during startup with error:
````
Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
InnoDB: preallocating 12582912 bytes for file ./ibtmp1 failed with error 38
InnoDB: Could not set the file size of './ibtmp1'. Probably out of disk space
InnoDB: Unable to create the shared innodb_temporary
InnoDB: Plugin initialization aborted with error Generic error
````

**Comment by ebriney** [comment]
Hi @strarsis, looks like a different error, do you have a simple compose file to reproduce it?

**Comment by stephen-turner** [comment]
Closing this ticket as the original issue is fixed in 2.3. @strarsis if you are still having problems with the other issue, please open a new ticket. Thanks.

**Comment by strarsis** [comment]
@ebriney: Sorry, had no time for creating some reproducible. I create a new issue and provide further information if needed.
